### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by git-elliot**
Bug fixes and GHA for flutter
1. This should slow our problem of not able to send resource intensive object like ActiveHost in spawned isolates https://github.com/rmawatson/flutter_isolate/issues/137.
2. But this will resolve hostname, mdnsInfo, and deviceName in main isolate.
3. Flutter checks will only run on dev-flutter branch for network_tools_flutter path match.

### CODE REVIEW

File: .github/workflows/flutter.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
     branches: [ "main" ]
     paths: ['network_tools_flutter/**']
   pull_request:
-    branches: [ "main", "dev" ]
+    branches: [ "main", "dev-flutter" ]
```
Comment (guyluz11) [inline .github/workflows/flutter.yml:13]: We are not removing dev so there is no reason to remove it from here

Comment (git-elliot) [reply .github/workflows/flutter.yml:13]: I want to run flutter based checks on dev-flutter and dart based checks on dev

Comment (git-elliot) [reply .github/workflows/flutter.yml:13]: And also if you check logs for GHA steps we can no more send resource intensive objects in message passing to isolates. Check below error.

Invalid argument(s): Illegal argument in isolate message: object is unsendable - Library:'dart:async' Class: _Future@4048458 (see restrictions listed at `SendPort.send()` documentation for more information)

Comment (guyluz11) [reply .github/workflows/flutter.yml:13]: > 

Ho ok, so this is ci file for flutter only

**Pull request merged**